### PR TITLE
308 ohe bugfix

### PR DESCRIPTION
**Pull request by alphanumericmale**
PR for bugfix for OHE transformer class. This update aims to allow for the acceptance of sklearn _drop_idx_after_grouping attribute. This will in turn allow OHE to be used with versions of sklearn > 1.2.1

Issue here: https://github.com/lvgig/tubular/issues/308

### CODE REVIEW

**Comment by davidhopkinson26** [comment]
Have just noted that running tests on main with sklearn 1.3.2 does not seem to fail which suggest the issue encountered in #308 is not as reproducible as first thought.

File: tests/nominal/test_OneHotEncodingTransformer.py [hunk 1]
```diff
@@ -264,3 +265,76 @@ def test_unseen_categories_encoded_as_all_zeroes(self, df_test, expected):
...
+    @pytest.fixture
+    def sample_data(self):
+        return pd.DataFrame(
+            {
+                "color": ["red", "blue", "green", "blue", "red"],
+                "size": ["S", "M", "L", "S", "M"],
+            },
+        )
+
+    def test_not_fitted_error_transform(self, sample_data):
```
Comment (davidhopkinson26) [inline tests/nominal/test_OneHotEncodingTransformer.py:282]: duplicate of GenericNominalTransformTests.test_not_fitted_error_raised

File: tests/nominal/test_OneHotEncodingTransformer.py [hunk 2]
```diff
@@ -264,3 +265,76 @@ def test_unseen_categories_encoded_as_all_zeroes(self, df_test, expected):
...
+
+    def test_initialization_and_fitting(self, sample_data):
+        transformer = OneHotEncodingTransformer(columns=["color", "size"])
+        transformer.fit(sample_data)
+        assert hasattr(
+            transformer._encoder,
+            "categories_",
+        ), "The encoder should have 'categories_' attribute after fitting."
+
+    def test_transformation(self, sample_data):
```
Comment (davidhopkinson26) [inline tests/nominal/test_OneHotEncodingTransformer.py:295]: We already test outputs of transform in TestTransform.  Don't think this is adding anything new?  if it is then it should be in that test class with amore descriptive name.

File: tests/nominal/test_OneHotEncodingTransformer.py [hunk 3]
```diff
@@ -264,3 +265,76 @@ def test_unseen_categories_encoded_as_all_zeroes(self, df_test, expected):
...
+            "color_green",
+            "size_S",
+            "size_M",
+            "size_L",
+        ]
+        assert all(
+            col in transformed.columns for col in expected_columns
+        ), "All expected columns should be in the transformed dataframe."
+
+    def test_unseen_categories(self, sample_data):
```
Comment (davidhopkinson26) [inline tests/nominal/test_OneHotEncodingTransformer.py:311]: again, duplication of tests in TestTransform

File: tests/nominal/test_OneHotEncodingTransformer.py [hunk 4]
```diff
@@ -264,3 +265,76 @@ def test_unseen_categories_encoded_as_all_zeroes(self, df_test, expected):
...
+            "color_green",
+            "size_S",
+            "size_M",
+            "size_L",
+        ]
+        assert all(
+            col in transformed.columns for col in expected_columns
+        ), "All expected columns should be in the transformed dataframe."
+
+    def test_bug_fix(self, sample_data):
```
Comment (davidhopkinson26) [inline tests/nominal/test_OneHotEncodingTransformer.py:334]: I think this is a duplicate as well

**Review by davidhopkinson26 — CHANGES_REQUESTED** [review]
Despite not being sure about the necessity of this change I think the set up is more sensible than what we had previously (direct calls to OneHotEncoder methods).  Arguably a still better approach might be to inherit from OnehotEncoder but that does risk some complex interactions form multiinheritance.

I do not think the additional tests you have added are needed though.

File: tests/nominal/test_OneHotEncodingTransformer.py (hunk 1 above)
Comment (alphanumericmale) [reply tests/nominal/test_OneHotEncodingTransformer.py:282]: removed

File: tests/nominal/test_OneHotEncodingTransformer.py (hunk 3 above)
Comment (alphanumericmale) [reply tests/nominal/test_OneHotEncodingTransformer.py:311]: remove

File: tests/nominal/test_OneHotEncodingTransformer.py (hunk 2 above)
Comment (alphanumericmale) [reply tests/nominal/test_OneHotEncodingTransformer.py:295]: removed

File: tests/nominal/test_OneHotEncodingTransformer.py [hunk 5]
```diff
@@ -264,3 +265,76 @@ def test_unseen_categories_encoded_as_all_zeroes(self, df_test, expected):
...
+                "size": ["S", "M", "L", "S", "M"],
+            },
+        )
+
+    def test_not_fitted_error_transform(self, sample_data):
+        transformer = OneHotEncodingTransformer(columns=["color", "size"])
+        with pytest.raises(NotFittedError):
+            transformer.transform(sample_data)
+
+    def test_initialization_and_fitting(self, sample_data):
```
[the remark this answers is not in the document]
Comment (alphanumericmale) [reply tests/nominal/test_OneHotEncodingTransformer.py:287]: removed

File: tests/nominal/test_OneHotEncodingTransformer.py (hunk 4 above)
Comment (alphanumericmale) [reply tests/nominal/test_OneHotEncodingTransformer.py:334]: removed

**Review by davidhopkinson26 — APPROVED** [review]
Thanks @alphanumericmale!

**Review by davidhopkinson26 — CHANGES_REQUESTED** [review]
One last thing - could you update the changelog please?

**Review by davidhopkinson26 — APPROVED** [review]
thanks!

**Pull request merged**